e2e/fbproxy/app/proxy.go

```diff
@@ -58,11 +58,29 @@ func (p *proxy) Proxy(w http.ResponseWriter, r *http.Request) {
 
 	err := p.proxy(ctx, w, r)
 	if err != nil {
+		if isContextCanceled(err) {
+			http.Error(w, "proxy closed", http.StatusServiceUnavailable)
+			return
+		}
+
 		log.Error(ctx, "Proxy error", err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
 
+// isContextCanceled checks if the error or any of its wrapped errors is context.Canceled.
+func isContextCanceled(err error) bool {
+	for {
+		if err == nil {
+			return false
+		}
+		if errors.Is(err, context.Canceled) {
+			return true
+		}
+		err = errors.Unwrap(err)
+	}
+}
```

Comment:
`errors.Is(err, context.Canceled)` should be sufficient, it does check the whole chain, no need to `Unwrap` yourself.